internal/model/segment.go

```diff
@@ -0,0 +1,37 @@
+package model
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type Segment struct {
+	ID                string `json:"id,omitempty"`
```

Comment:
Just curious, why is it ok if this is empty, shouldn't there always be an ID?
Might be better to add a comment explaining why, instead of replying on my comment.

Reply:
The only time it's empty is on a `Create` call, since it's filled in by the server. I don't mind removing the omitempty, it's ignored by create either way.